ols/src/query_helpers/docs_summarizer.py

```diff
@@ -218,68 +240,76 @@ async def iterate_with_tools(
         Yields:
             StreamedChunk objects representing parts of the response
         """
+        mcp_servers = self.mcp_config_to_client_dump()
         for i in range(1, max_rounds + 1):
-            is_final_round = (not self._introspection_enabled) or (i == max_rounds)
-            tool_call_chunks = []
-            # invoke LLM and process response chunks
-            async for chunk in self._invoke_llm(
-                messages,
-                llm_input_values,
-                tools_map=tools_map,
-                is_final_round=is_final_round,
-                token_counter=token_counter,
-            ):
-                # TODO: Temporary fix for fake-llm (load test) which gives
-                # output as string. Currently every method that we use gives us
-                # proper output, except fake-llm. We need to move to a different
-                # fake-llm (or custom fake-llm) which can handle streaming/non-streaming
-                # & tool calling and gives response not as string. Even below
-                # temp fix will fail for tool calling.
-                # (load test can be run with tool calling set to False till we
-                # have a permanent fix)
-                if isinstance(chunk, str):
-                    yield StreamedChunk(type="text", text=chunk)
-                    break
-
-                # check if LLM has finished generating
-                if chunk.response_metadata.get("finish_reason") == "stop":  # type: ignore [attr-defined]
-                    return
+            async with asyncio.timeout(constants.TOOL_CALL_ROUND_TIMEOUT):
```

Comment:
I've added this timeout because I forgot the
```python
if __name__ == "__main__":
    mcp.run(transport="stdio")
```
in the mcp/openshift tools and was scratching my head for a while why it is not doing anything and waiting forever.
But it is TBD. I see some value in limiting the rounds, but there is some line between beying stuck and executing many/long running tools.
I'm also not sure how that would work with human in the loop. So I'm fine with keeping this while it has value, or removing now.